scripts/build/builders/nxp/builder.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+from ..gn import GnBuilder
+
+
+def check_modifiers(modifiers, supported_modifiers):
+    """Checks that every modifier is a key in the supported modifiers dictionary.
+
+    If a modifier is not present in the dictionary, it most likely means a gn generation error.
+    """
+    if not supported_modifiers:
+        print((
+            'A list with supported variable modifiers is not specified.\n'
+            'Either provide one or do not use --variable-modifiers option in build_examples command.'
+        ))
+        raise ValueError
+
+    supported_ids = [modifier['id'] for modifier in supported_modifiers]
+    ids = [modifier.split('=')[0] for modifier in modifiers.split(' ')]
+    for id_ in ids:
+        if id_ not in supported_ids:
+            print(f'"{id_}" arg is not supported. Please check the supported list: {supported_ids}')
+            raise ValueError
+
+    return True
+
+
+def generate_option_from_modifiers(modifiers, supported_modifiers):
+    """Generates an aggregated gn option by replacing the modifier target name with the actual gn option.
+
+    For example, if the modifiers is a string with the following value "var-mod=foo another-var-mod=bar" and
+    the platform JSON file contains a "variable_modifiers" list entry:
```

Comment:
This feels very free-form. In your PR description, could you describe what needs are known *today*?

Overall I would like to apply some `You are not going to need it` here and only implement what you actually need. While reviewing this, going through free-form dictionaries and lists seems overly complex. I don't think we need it and it makes the builders overall hard to maintain.